ra-optimisation/linear-rao/src/main/java/com/farao_community/farao/linear_rao/Situation.java

```diff
@@ -0,0 +1,188 @@
+            crac.addExtension(ResultVariantManager.class, resultVariantManager);
+        }
+
+        String situationVariantId = getUniqueSituationId();
+        network.getVariantManager().cloneVariant(network.getVariantManager().getWorkingVariantId(), situationVariantId);
+        crac.getExtension(ResultVariantManager.class).createVariant(situationVariantId);
+        variantIds.add(situationVariantId);
+        setWorkingVariant(situationVariantId);
+        init();
+
+        this.systematicSensitivityAnalysisResultMap = new HashMap<>();
+    }
+
+    /**
+     * Add in the Crac extension the initial RangeActions set-points
+     */
+    private void init() {
+        String preventiveState = getCrac().getPreventiveState().getId();
+        for (RangeAction rangeAction : getCrac().getRangeActions()) {
+            double valueInNetwork = rangeAction.getCurrentValue(network);
+            RangeActionResultExtension rangeActionResultMap = rangeAction.getExtension(RangeActionResultExtension.class);
+            RangeActionResult rangeActionResult = rangeActionResultMap.getVariant(workingVariantId);
+            rangeActionResult.setSetPoint(preventiveState, valueInNetwork);
+            if (rangeAction instanceof PstRange) {
+                ((PstRangeResult) rangeActionResult).setTap(preventiveState, ((PstRange) rangeAction).computeTapPosition(valueInNetwork));
+            }
+        }
+    }
+
+    /**
+     * get a unique network variant ID
+     */
+    private String getUniqueNetworkVariantId() {
+        String uniqueId;
+        do {
+            uniqueId = UUID.randomUUID().toString();
+        } while (network.getVariantManager().getVariantIds().contains(uniqueId));
+        return uniqueId;
+    }
+
+    private String getUniqueSituationId() {
+        String networkVariantId = getUniqueNetworkVariantId();
+        String cracVariantId = crac.getExtension(ResultVariantManager.class).getUniqueVariantId();
+        return String.valueOf(networkVariantId.concat(cracVariantId).hashCode());
+    }
+
+    /**
+     * Compare the network situations (i.e. the RangeActions set-points) of two
+     * AbstractSituation. Returns true if the situations are identical, and false
+     * if they are not.
+     */
+    boolean sameRemedialActions(String id1, String id2) {
+        String preventiveState = crac.getPreventiveState().getId();
+        for (RangeAction rangeAction : crac.getRangeActions()) {
+            RangeActionResultExtension rangeActionResultMap = rangeAction.getExtension(RangeActionResultExtension.class);
+            double value1 = rangeActionResultMap.getVariant(id1).getSetPoint(preventiveState);
+            double value2 = rangeActionResultMap.getVariant(id2).getSetPoint(preventiveState);
+            if (value1 != value2 && (!Double.isNaN(value1) || !Double.isNaN(value2))) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    String cloneVariant(String referenceVariantId) {
```

Comment:
maybe rename this method since it only clones the network but creates a fresh result variant

Reply:
I'm not sure how to rename it